avocado/core/test.py

```diff
@@ -1093,13 +1094,40 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
         :param locations: list of URLs from where the asset can be
                           fetched (optional)
         :param expire: time for the asset to expire
-        :raise EnvironmentError: When it fails to fetch the asset
-        :returns: asset file local path
+        :param find_only: whether it should only look for the asset in the
+                          cache and do not download/move it to the cache.
+        :param cancel_on_missing: whether the test should be canceled if the
+                                  asset was not found in the cache or if
+                                  `fetch` could not add the asset to the cache.
+        :raises: :class:`exceptions.TestCancel when `cancel_on_missing` is
+                 `True` and the asset was not found in the cache or it could
+                 not be added to the cache.
+        :raises OSError: when it fails to fetch the asset and
+                         `cancel_on_missing` is `False`.
+        :returns: asset file local path or `None` when asset was not found in
+                  the cache.
         """
         if expire is not None:
             expire = data_structures.time_to_seconds(str(expire))
-        return asset.Asset(name, asset_hash, algorithm, locations,
-                           self.cache_dirs, expire).fetch()
+
+        asset_obj = asset.Asset(name, asset_hash, algorithm, locations,
+                                self.cache_dirs, expire)
+
+        missing_asset_message = 'Missing asset ' + name
```

Comment:
String formatting is preferable to concatenations.

Reply:
Changed on v3!